src/Storages/ObjectStorage/StorageObjectStorageSource.cpp

```diff
@@ -135,11 +137,22 @@ std::shared_ptr<StorageObjectStorageSource::IIterator> StorageObjectStorageSourc
     std::unique_ptr<IIterator> iterator;
     if (configuration->isPathWithGlobs())
     {
-        /// Iterate through disclosed globs and make a source for each file
-        iterator = std::make_unique<GlobIterator>(
-            object_storage, configuration, predicate, virtual_columns,
-            local_context, is_archive ? nullptr : read_keys, query_settings.list_object_keys_size,
-            query_settings.throw_on_zero_files_match, file_progress_callback);
+        if (configuration->pathHasExactlyOneBracketsExpansion())
```

Comment:
But we do not need "exactly one" condition here. We may expand as many selection (enum) globs as we like.

This is how it worked prior to being lost

Reply:
Also, `expandSelectorGlobImpl()` will expand all existing selection globs (not only one).
What needs to be checked is that it handles a selection-with-no-selection properly (`{onlyoption}`)

Reply:
Yes, I got it a bit later in the day yesterday, we don't need to limit it to just one instance. I'll remove it

A bit disappointing though that `expandSelectorGlobImpl` does not unwrap ranges, so I'd have to make a second iteration later.

Reply:
The case I'm trying to fix had exactly one enum expanding, so I decided go safer and enable multiple glob expansion later, as [I already broke something](https://github.com/ClickHouse/ClickHouse/pull/73569) last time I touched `parseGlobs.cpp`.

Reply:
Unwrapping ranges can be a bit more dangerous -- e.g. it may lead to very large set of resulting paths (`prefix/option{1..1000000}`), maybe this is OK actually

Reply:
Well yeah, it is pretty easy to make it explode, we can decide during unwrapping that if we already have more than `MaxKeys`, we should fall back to listing everything.